### ISSUE TITLE
Keep track of HHH-14344 about insert statement  ordering issue

### ISSUE THREAD

**Issue by DavideD** [hibernate/hibernate-reactive#561]
See this potential fix: https://github.com/hibernate/hibernate-orm/pull/3660#

I haven't looked at the details, but we might want to adapt the test for reactive and review the ActionQueue class.

**Comment by gavinking** [comment]
Yes, thanks @DavideD , we do need to keep track of this. I’m really confused as to how we managed to get out of sync with core here. Can’t even tell if it was intentional or accidental.
